heron/statefulstorages/src/java/org/apache/heron/statefulstorage/localfs/LocalFileSystemStorage.java

```diff
@@ -134,6 +146,23 @@ public void dispose(String topologyName, String oldestCheckpointPreserved,
     }
   }
 
+  private void cleanCheckpoints(String path, int remaining) throws StatefulStorageException {
+    if (FileUtils.isDirectoryExists(path) && FileUtils.hasChildren(path)) {
+      File[] children = new File(path).listFiles();
```

Comment:
suggest to keep it consistent with dispose(), which uses:

String[] names = new File(topologyCheckpointRoot).list();

Sorting might be cheaper this way as well.

Reply:
done